### PR TITLE
feat(core): [STLX-4305] add feeLimit parameter to Send options

### PR DESCRIPTION
**Pull request by matilareo**
stlx-4309

## Changes 
- add feeLimit parameter to Send Options on core

### CODE REVIEW

**Review by fmlenarduzzi — APPROVED** [review]

File: modules/core/src/v2/wallet.ts [hunk 1]
```diff
@@ -1981,7 +1983,9 @@ export class Wallet {
     if (params.data && coin.transactionDataAllowed()) {
       recipients[0].data = params.data;
     }
-
+    if (coin.getFamily()==='trx' && params.feeLimit) {
```
Comment (twtaylorbitgo) [inline modules/core/src/v2/wallet.ts:1986]: spacing

**Review by twtaylorbitgo — CHANGES_REQUESTED** [review]

**Review by fmlenarduzzi — DISMISSED** [review]

**Review by mcasares-simtlix — DISMISSED** [review]

**Review by twtaylorbitgo — APPROVED** [review]

**Review by twtaylorbitgo — CHANGES_REQUESTED** [review]
This needs to be on build and send transaction in BitGoJS

Also this should be called `maxFee` - this is a parameter we should re-use in WP

**Review by karankurbur — DISMISSED** [review]

**Comment by matilareo** [comment]
> This needs to be on build and send transaction in BitGoJS
> 
> Also this should be called `maxFee` - this is a parameter we should re-use in WP

The method that i've modified is doing both, build and send. 
look at these lines:

- https://github.com/BitGo/BitGoJS/blob/7f6ac94eab0d62805a5265bfba6f582a67339593/modules/core/src/v2/wallet.ts#L1990
- https://github.com/BitGo/BitGoJS/blob/7f6ac94eab0d62805a5265bfba6f582a67339593/modules/core/src/v2/wallet.ts#L2047
- https://github.com/BitGo/BitGoJS/blob/7f6ac94eab0d62805a5265bfba6f582a67339593/modules/core/src/v2/wallet.ts#L1807
- https://github.com/BitGo/BitGoJS/blob/7f6ac94eab0d62805a5265bfba6f582a67339593/modules/core/src/v2/wallet.ts#L1682

**Review by CapnMigraine — APPROVED** [review]

**Review by tylerlevine — CHANGES_REQUESTED** [review]

File: modules/core/src/v2/wallet.ts [hunk 2]
```diff
@@ -352,6 +353,7 @@ export interface SendManyOptions {
   recipients?: {
     address: string;
     amount: string | number;
+    feeLimit?: string;
```
Comment (tylerlevine) [inline modules/core/src/v2/wallet.ts:356]: Why is this option specified per-recipient?

File: modules/core/src/v2/wallet.ts [hunk 3]
```diff
@@ -1981,7 +1983,9 @@ export class Wallet {
     if (params.data && coin.transactionDataAllowed()) {
       recipients[0].data = params.data;
     }
-
+    if (coin.getFamily() === 'trx' && params.feeLimit) {
```
[the remark this answers is not in the document]
Comment (matilareo) [reply modules/core/src/v2/wallet.ts:1986]: you are right, thank you

File: modules/core/src/v2/wallet.ts (hunk 2 above)
Comment (matilareo) [reply modules/core/src/v2/wallet.ts:356]: this is because each contract call could have different fee limit

**Review by fmlenarduzzi — APPROVED** [review]

**Review by karankurbur — APPROVED** [review]

File: modules/core/src/v2/coins/trx.ts [hunk 4]
```diff
@@ -312,6 +312,26 @@ export class Trx extends BaseCoin {
...
 
+  /**
+   * Modify prebuild before sending it to the server.
+   * @param buildParams The whitelisted parameters for this prebuild
+   * @param callback
+   */
+  getExtraPrebuildParams(buildParams: any, callback?: NodeCallback<any>): Bluebird<any> {
+    const recipients: any[] = [{}];
+    return co<any>(function* () {
+      if (buildParams.recipients[0].data && buildParams.feeLimit) {
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/trx.ts:323]: is only one recipient supported?

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**